### PR TITLE
Update mnemonic addresses page

### PR DESCRIPTION
**Pull request by IViol**
JWWEB-98

### CODE REVIEW

File: src/utils/wallets/checkMnemonicType.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// @flow
+
+function checkMnemonicType(walletType: ?WalletType): boolean {
```
Comment (flexxnn) [inline src/utils/wallets/checkMnemonicType.js]: can be ranamed to checkTypeIsMnemonic

File: src/services/keystore.js [hunk 2]
```diff
@@ -460,6 +460,38 @@ function getBackupData(wallets: Wallets, walletId: string, password: string): st
...
+    address,
+    bip32XPublicKey,
+    addressIndex,
+  }: Wallet = getWallet(wallets, walletId)
+
+  if (type === ADDRESS_TYPE) {
+    return address
+  }
+
+  const indexEnd: number = (addressIndex || 0) + 1
```
Comment (Romanior) [inline src/services/keystore.js]: again magic numbers why not + 2?

File: src/routes/Wallets/routes/Addresses/sagas/walletsAddresses.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  const { iteration } = yield select(selectWalletsAddresses)
+
+  const startIndex: Index = config.mnemonicAddressesCount * iteration
+  const endIndex: Index = (startIndex + config.mnemonicAddressesCount) - 1
+
+  if (!activeWalletId) {
+    return
+  }
+
+  yield put(walletsAddresses.getMoreRequest(items, activeWalletId, startIndex, endIndex))
```
Comment (Romanior) [inline src/routes/Wallets/routes/Addresses/sagas/walletsAddresses.js]: Ok, I understood the problem with this, should be like this

```suggestion
  if (activeWalletId) {
    return  yield put(walletsAddresses.getMoreRequest(items, activeWalletId, startIndex, endIndex))
    }
```

File: src/services/keystore.js [hunk 4]
```diff
@@ -460,6 +460,38 @@ function getBackupData(wallets: Wallets, walletId: string, password: string): st
...
+
+  return bip32XPublicKey
+    ? utils.generateAddresses(bip32XPublicKey, addressIndex, indexEnd)[0]
+    : null
+}
```
Comment (Romanior) [inline src/services/keystore.js]:
```javascript
function getAddress(wallets: Wallets, walletId: string): ?Address {
  const {
          type,
          address,
          bip32XPublicKey,
          addressIndex,
        }: Wallet = getWallet(wallets, walletId);

  if (type === ADDRESS_TYPE) {
    return address
  }
  
  if (bip32XPublicKey) {
    const indexEnd: number = (addressIndex || 0) + 1;
    return utils.generateAddresses(bip32XPublicKey, addressIndex, indexEnd)[0];
  }
}

```

**Pull request merged**